Assignment2/exercise1/BinaryTree.java

```diff
@@ -0,0 +1,38 @@
+package exercise1;
+
+public class BinaryTree<Integer>{
+  private Node<Integer> root;
+  private int size;
+  
+  public BinaryTree(){
+    size=0;
+    root=null;
+  }
+  
+  public BinaryTree(Node<Integer> root){
+    size=1;
+    this.root=root;
+  }
+  
+  public void insert(Node<Integer> parent, int way, Node<Integer> newNode){
+    if (parent != null) {
+      if (way == 0) parent.setLeft(newNode);
+      else parent.setRight(newNode);
+    }
+  }
+  
+  public boolean printAncestors(Node<Integer> node, int key){
```

Comment:
This is nice code. However, it isn't necessarily the most readable (having a return value from a print method isn't obvious what it does). Maybe you could add some javadoc to explain the return type.